confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java

```diff
@@ -87,33 +89,80 @@ public int getDeviceGroupID(String device) {
   }
 
   /** Build ConfigNodeGroup ConsensusLayer */
-  private void setConsensusLayer(ConfigNodeConf config) {
-    // TODO: Support other consensus protocol
-    this.consensusImpl = new StandAloneConsensus(id -> new PartitionRegionStateMachine());
-    // TODO: handle the possible exception that cause the Consensus fail
-    try {
-      this.consensusImpl.start();
-    } catch (IOException e) {
-      e.printStackTrace();
+  private void setConsensusLayer() throws IOException {
+    // There is only one ConfigNodeGroup
+    consensusGroupId = new ConsensusGroupId(GroupType.PartitionRegion, 0);
+
+    // If consensusDir does not exist, create consensusDir
+    File consensusDir = new File(conf.getConsensusDir());
+    if (!consensusDir.exists()) {
+      if (consensusDir.mkdirs()) {
+        LOGGER.info("Make consensus dirs: {}", consensusDir);
+      } else {
+        throw new IOException(
+            String.format(
+                "Start ConfigNode failed, because couldn't make system dirs: %s.",
+                consensusDir.getAbsolutePath()));
+      }
     }
 
-    this.consensusGroupId = new ConsensusGroupId(GroupType.PartitionRegion, 0);
-    this.consensusImpl.addConsensusGroup(
-        this.consensusGroupId,
+    // Implement specific consensus
+    switch (conf.getConsensusType()) {
+      case STANDALONE:
+        constructStandAloneConsensus();
+        break;
+      case RATIS:
+        constructRatisConsensus();
```

Comment:
looks like the constructors of these two consensus are not consistent

Reply:
Yes, because they construct different consensus protocol.